simulation.cpp

```diff
@@ -22,10 +22,13 @@ simulation::simulation(int init_pop_size,
     m_optimal_output{1}
 {
     m_rng.seed(m_seed);
-    for(auto& ind : m_population.get_inds())
-    {
-        ind = individual{net_param{net_arch}};
+
+    std::vector<individual> vec_inds;
+    for(size_t i = 0; i != m_population.get_inds().size(); ++i){
+       individual ind = individual{net_param{net_arch}};
+       vec_inds.push_back(ind);
     }
+    m_population.change_vector_individuals(vec_inds);
```

Comment:
I think this is not really necessary, here you jsut need to put the rightly contructed individuals inside a population of already the correct size, so:
```suggestion
    for(size_t i = 0; i != m_population.get_inds().size(); ++i){
      m_population.change_nth_ind(i, individual{net_param{net_arch}});
    }
```

Reply:
I am not sure this is the correct syntax for the funciton, but you see what I mean.